### ISSUE TITLE
Exception in Semantics library running Jim's demo in the iOS simulator

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#4150]
I'm hitting this when running @jimbeveridge's demo.  Previously I was seeing it crash the iOS port with a null pointer in c++ code, but I think that's now fixed and we're just hitting a non-fatal Dart exception.

I can provide you the source code if needed.

@Hixie @tvolkert 

```
══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
The following assertion was thrown during a scheduler callback:
'package:flutter/src/rendering/semantics.dart': Failed assertion: line 414 pos 14:
'node.parent?._dirty != true' is not true.
Either the assertion indicates an error in the framework itself, or we should provide substantially
more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new
When the exception was thrown, this was the stack:
#0      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:27)
#1      SemanticsNode.sendSemanticsTree (package:flutter/src/rendering/semantics.dart:414:14)
#2      BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding.beginFrame (package:flutter/src/rendering/binding.dart:158:21)
#3      BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding&WidgetsBinding.beginFrame (package:flutter/src/widgets/binding.dart:205:21)
#4      BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:144:5)
#5      BindingBase&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:423:15)
#6      BindingBase&SchedulerBinding.handleBeginFrame (package:flutter/src/scheduler/binding.dart:383:7)
#7      _beginFrame (file:///b/build/slave/Mac_Engine/build/src/out/ios_sim_Release/gen/sky/bindings/hooks.dart:48)
════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by Hixie** [comment]
Source code and steps to reproduce would be very helpful yes.

**Comment by abarth** [comment]
People keep hitting this on gitter

**Comment by sethladd** [comment]
Are we still waiting on a way to repro this, or can we repro this ourselves?

**Comment by eseidelGoogle** [comment]
Gitter: https://gitter.im/flutter/flutter?at=58333a1ac790e0ec5613d745

Note the Gitter report was from linux/android not iOS like my original comment.

**Comment by zoechi** [comment]
https://github.com/zoechi/flutter_myapp in the issue#4150 branch 
Open the drawer with the top-left icon-button and click `Account` menu item.

**Comment by zoechi** [comment]
I get this exception also with the stocks example when I click on any stock item in the main screen.

**Comment by zoechi** [comment]
I copied the files into a project created with `flutter create` because the example was not included when I checked out the `flutter` repo.

**Comment by sethladd** [comment]
Does this still happen?

**Comment by Hixie** [comment]
Likely dupe of https://github.com/flutter/flutter/issues/7287.

**Comment by goderbauer** [comment]
I can't reproduce this anymore. This code has also changed a bit in the last few weeks. If you still see this, feel free to open a new bug with steps to reproduce.
